src/nvim/option.c

```diff
@@ -1442,15 +1442,18 @@ do_set (
               goto skip;
             }
 
-            if (adding)
-              value = *(long *)varp + value;
-            if (prepending)
-              value = *(long *)varp * value;
-            if (removing)
-              value = *(long *)varp - value;
-            errmsg = set_num_option(opt_idx, varp, value,
-                errbuf, sizeof(errbuf), opt_flags);
-          } else if (opt_idx >= 0) {                      /* string */
+            if (adding) {
+              value = *(intmax_t *)varp + value;
```

Comment:
same issue here (`varp` is `NumOpt*`). I wonder if it's not cleaner to let `value` be `NumOpt` and just have a `intmax_t` temporary just around `vim_str2nr`. Also these are exclusive so use `else if` between cases.